priv/priv.go

```diff
@@ -67,21 +69,28 @@ func (m *Mutex) commonChecks() error {
 // Lock attempts to acquire the mutex lock, and wil block if it is
 // already locked.
 func (m *Mutex) Lock() error {
-	if err := m.commonChecks(); err != nil {
-		return err
-	}
-
-	return m.lock.Lock(true)
+	return m.lock(true)
 }
 
 // TryLock attempts to acquire the mutex lock. If it is already locked,
 // it will return ErrAlreadyLocked.
 func (m *Mutex) TryLock() error {
+	return m.lock(false)
+}
+
+func (m *Mutex) lock(blocking bool) error {
 	if err := m.commonChecks(); err != nil {
```

Comment:
Do we need a package for that one line doing uid() == 0?  Feels too trivial for that. Or do I misunderstand what you mean?